src/core/Akka.Persistence/SnapshotProtocol.cs

```diff
@@ -52,32 +52,28 @@ public int Compare(SnapshotMetadata x, SnapshotMetadata y)
         }
 
         /// <summary>
-        /// TBD
+        /// The singleton comparer instance.
         /// </summary>
         public static IComparer<SnapshotMetadata> Comparer { get; } = new SnapshotMetadataComparer();
 
-        /// <summary>
-        /// TBD
-        /// </summary>
-        public static DateTime TimestampNotSpecified = DateTime.MinValue;
-
         /// <summary>
         /// Initializes a new instance of the <see cref="SnapshotMetadata"/> class.
         /// </summary>
         /// <param name="persistenceId">The id of the persistent actor fro which the snapshot was taken.</param>
         /// <param name="sequenceNr">The sequence number at which the snapshot was taken.</param>
+        [Obsolete("This constructor is deprecated and will be removed in v1.6. Use the constructor with the timestamp parameter instead.", true)]
         public SnapshotMetadata(string persistenceId, long sequenceNr)
-            : this(persistenceId, sequenceNr, TimestampNotSpecified)
+            : this(persistenceId, sequenceNr, DateTime.UtcNow)
```

Comment:
The only objection I have for this is that this breaks previous behaviour.
The way this works previously, if `DateTime.MinValue` were passed as the timestamp, it will be interpreted as a `null` value by the `SqlSnapshotStore`, and if its a null, it will not generate a timestamp SQL query statement inside the `QueryExecutor`

If we're passing `DateTime.UtcNow` here, we're basically changing the `QueryExecutor` SQL query from `DeleteSnapshotSql` to `DeleteSnapshotRangeSql + " AND { Configuration.TimestampColumnName} = @Timestamp"`

Reply:
That's the point of this change though - we need to be passing a real timestamp value. I deprecated this CTOR and we aren't using it anymore - we're using the timestamp provided by the `ITimeProvider` in the 3-parameter CTOR.